forum/qa-plugin/report-reason/script.js

```diff
@@ -0,0 +1,116 @@
+document.addEventListener("DOMContentLoaded", () => {
+    
+    const flagboxPopup = document.querySelector("#flagbox-popup");
+    const wrap = document.querySelector(".qa-flag-reasons-wrap");
+    const body = document.body;
+    const closer = document.querySelector(".close-preview-btn");
+    const sendButton = document.querySelector(".qa-go-flag-send-button");
+    const errorMessage = "Błąd serwera. Proszę spróbować za jakiś czas";
+    const errorPopup = document.querySelector("#qa-spam-reason-error");
+    const tooManyReportError = "Zbyt dużo zgłoszeń z tego adresu IP. Spróbuj za godzinę";
+    const reportReasonEmptyError = "Wybierz powód zgłoszenia!";
+    const commentSubmitButton = document.querySelector('.qa-form-tall-button-comment')
+
+    function showPopup() {
+        flagboxPopup.classList.remove("hide");
+        flagboxPopup.classList.add("show");
+    }
+    function hidePopup() {
+        flagboxPopup.classList.remove("show");
+        flagboxPopup.classList.add("hide");
+    }
+    function removeAllChildFromErrorPopup() {
+        while (errorPopup.firstChild) {
+             errorPopup.removeChild(errorPopup.firstChild);
+        }
+    }
+    function showError(errorText) {
+        const paragraph = document.createElement("p");
+        paragraph.textContent = errorText;
+                    
+        removeAllChildFromErrorPopup();
+        errorPopup.appendChild(paragraph);
+        errorPopup.hidden = false
+    }
+
+    wrap.addEventListener("click", (e) => {
+        e.stopPropagation();
+    });
+    
+    commentSubmitButton.addEventListener("click", () => {
+       location.reload(); 
+    });
+    
+    body.addEventListener("click", (event) => {
+        if (event.target.matches(".qa-form-light-button-unflag")) {
+            location.reload();
+        }
+        if (event.target.matches(".qa-form-light-button-clearflags")) {
+            location.reload();
+        }
+        if (event.target.matches("input.qa-form-light-button-flag")) {
+            const flagButton = event.target;
+            flagButton.type = "button";
+            
+            event.preventDefault();
+            showPopup();    
+
+            flagboxPopup.addEventListener("click", () => {
+                hidePopup();
+            });
+
+            closer.addEventListener("click", () => {
+                hidePopup();
+                return false;
+            });
+
+            sendButton.addEventListener("click", () => {
+
+                const flagReason = document.querySelector("input.qa-spam-reason-radio:checked");
+                const flagNotice = document.querySelector(".qa-spam-reason-text").value;
+
+                if (!flagReason) {
+                    showError(reportReasonEmptyError);
+                    return false;
+                }
+
+                const { postid: postId, posttype: postType, parentid: parentId } = flagButton.dataset;
```

Comment:
Nie pamiętam jak stare przeglądarki wspieramy, ale spread operator [nie jest obsługiwany w IE11](https://caniuse.com/#feat=mdn-javascript_operators_spread_spread_in_destructuring).

Reply:
Nie widzę, żeby property `parentId` było gdzieś używana w kodzie - czy jest sens to rozpakowywać?